src/QueryEditor.tsx

```diff
@@ -197,12 +209,41 @@ export class QueryEditor extends PureComponent<Props, QueryEditorState> {
     };
   };
 
-  onWillApplySuggestion = (suggestion: string, state: SuggestionsState): string => {
-    const hasBracket = state.typeaheadText.includes('{');
+  /**
+   * Fired when a typeahead suggestion is selected. The return value of this
+   * method gets placed in the query.
+   * */
+  onWillApplySuggestion = (suggestion: string, suggestionsState: SuggestionsState): string => {
+    // Modify suggestion based on context
+    switch (suggestionsState.typeaheadContext) {
+      case metricsContext: {
+        this.refreshLabelSuggestions(suggestion);
+        this.setState({ selectedMetricName: suggestion });
+        break;
+      }
 
-    // User selected a metric name, so we refresh our labels
-    if (!hasBracket) {
-      this.onSelectMetric([suggestion]);
+      case labelsContext: {
+        const nextChar = DOMUtil.getNextCharacter();
```

Comment:
maybe we include a warning about domutil?

Reply:
Good callout; added!